src/main/java/com/klodnicki/Bike/v2/service/UserService.java

```diff
@@ -1,7 +1,68 @@
 package com.klodnicki.Bike.v2.service;
 
+import com.klodnicki.Bike.v2.DTO.user.UserForAdminResponseDTO;
+import com.klodnicki.Bike.v2.DTO.user.UserForNormalUserResponseDTO;
+import com.klodnicki.Bike.v2.model.entity.User;
+import com.klodnicki.Bike.v2.repository.UserRepository;
+import org.modelmapper.ModelMapper;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
 
+import java.util.ArrayList;
+import java.util.List;
+
 @Service
 public class UserService {
+
+    private final UserRepository userRepository;
+    private final ModelMapper modelMapper;
+
+    public UserService(UserRepository userRepository, ModelMapper modelMapper) {
+        this.userRepository = userRepository;
+        this.modelMapper = modelMapper;
+    }
+
+    public UserForAdminResponseDTO add(User user) {
+        User user1 = userRepository.save(user);
+
+        return modelMapper.map(user1, UserForAdminResponseDTO.class);
+    }
+
+    public List<UserForAdminResponseDTO> findAll() {
+        Iterable<User> users = userRepository.findAll();
+        List<UserForAdminResponseDTO> listUsersDTO = new ArrayList<>();
+
+        for (User user: users) {
+            UserForAdminResponseDTO userDto = modelMapper.map(user, UserForAdminResponseDTO.class);
+            listUsersDTO.add(userDto);
+        }
+        return listUsersDTO;
+    }
+
+    public UserForNormalUserResponseDTO findById(Long id) {
+        User user = getUser(id);
+
+        return modelMapper.map(user, UserForNormalUserResponseDTO.class);
+    }
+
+    private User getUser(Long id) {
+        return userRepository.findById(id).orElseThrow(IllegalArgumentException::new);
+    }
+
+    public ResponseEntity<?> banUser(Long id) {
+        User user = getUser(id);
+
+        user.setAccountValid(false);
+        userRepository.save(user);
+
+        if(user.isAccountValid()) {
```

Comment:
theoretically no, but it might be some occurrence and the method will not set the account as false and then what?
it is not a transactional operation yet, so it can send a false response that it has been done -> meanwhile it hasn't.